### PR TITLE
Updated code to comply with PHPStan level 7.

### PR DESCRIPTION
**Pull request by AlexSkrypnyk**
## Summary

Raises the PHPStan static-analysis level from 3 to 7 across the entire codebase, resolving all 180 violations that surfaced. The fixes span interfaces, concrete classes, context classes, manager classes, service container wiring, and test helpers. No public API behaviour has changed - every fix is a type annotation, return-type declaration, or narrowing cast that makes existing semantics explicit to the analyser.

### Config (`phpstan.neon`)
- Bumped `level` from `3` to `7`.
- Added `tests/behat/bootstrap/BehatCliTrait.php` to `excludePaths` because its only consumer (`BehatCliContext.php`) is already excluded and the trait cannot be analysed standalone.

### Interface return types
- `DrupalDriverManagerInterface` - added `void` and typed-array return types.
- `DrupalMailManagerInterface` - added `void`, `array`, and `?object` return types.
- `DrupalAwareInterface` - typed all setter/getter signatures.
- `DrupalAuthenticationManagerInterface`, `DrupalUserManagerInterface`, `FastLogoutInterface` - completed missing return-type declarations.

### Property typing
- Replaced untyped `protected $foo;` declarations with typed properties across all context classes (`RawDrupalContext`, `RawMailContext`, `DrupalContext`, `BatchContext`, `ConfigContext`, `MailContext`, `MessageContext`, `MinkContext`, `RandomContext`, `DrushContext`, `DrupalSubContextBase`).
- Added explicit `= []` or `= NULL` initialisers as required by PHPStan level 7.

### Type narrowing in context and manager classes
- `RawDrupalContext::nodeCreate()` / `termCreate()`: narrowed `object` PHPDoc return to `\stdClass`; added `instanceof \stdClass` runtime guard before callers access `->nid` / `->tid` / `->uid`.
- `DrupalContext`: replaced `if (!$user)` (flagged always-false from PHPDoc) with `if (!$user instanceof \stdClass)`.
- `DrupalAuthenticationManager::loggedIn()`: removed dead `if (!$page)` and replaced with `if ($page === NULL)` plus a targeted `@phpstan-ignore identical.alwaysFalse` for the test-mock case.

### Array and string return types
- Added generic-style PHPDoc hints (`array<int, \stdClass>`, `array<string, mixed>`, `array<int, string>`) throughout context classes and managers.
- Cast `string|false` returns from `glob()`, `getcwd()`, `fgets()`, `file_get_contents()` to `(string)` or applied `?: []` fallbacks.

### Hook reader / annotation reader
- Used `class-string<\Behat\Behat\Context\Context>` in PHPDoc to satisfy callable-array typing in `Annotation\Reader` and `Attribute\HookAttributeReader`.

### Service container and supporting classes
- `DrupalExtension` service container: added return types and typed array properties.
- `RegionSelector`, `DriverListener`, `DrupalParametersTrait`, `MinkAwareTrait`: added missing return types and typed properties.
- `BrowserKitFactory`: cast `string|false` from `getcwd()`.
- `scripts/docs.php`: cast `string|false` from `fgets()` / `file_get_contents()`.

### Test files
- `FeatureContext`: added typed properties and return types.
- All `tests/phpunit/src/*Test.php` files: annotated `createMockForIntersectionOfInterfaces()` returns with `@var` intersection types; added return types to helper methods.

## Before / After

```
Before                          After
────────────────────────────────────────────────────────────────────
PHPStan level    3              PHPStan level    7
Violations       180            Violations       0
```

```
Category of fix                                Files touched
┌──────────────────────────────────────────────┬────────────┐
│ Config (phpstan.neon)                        │     1      │
│ Interface return types                       │     6      │
│ Property typing (typed props + initialisers) │    11      │
│ Type narrowing (\stdClass guards, instanceof)│     3      │
│ Array / string return types & casts          │    12      │
│ Hook / annotation reader PHPDoc              │     2      │
│ Service container & support classes          │     7      │
│ Test files (PHPUnit + FeatureContext)         │    10      │
└──────────────────────────────────────────────┴────────────┘
Total                                               44 files
   428 insertions(+)  183 deletions(-)
```

## Summary by CodeRabbit

* **Refactor**
  * Enhanced codebase type safety and static analysis strictness, improving code reliability.
  * Added explicit return type declarations across core methods for better contract enforcement.
  * Strengthened property typing with more precise data structure documentation.

* **Bug Fixes**
  * Improved defensive coding practices with explicit type validation and null checks.
  * Enhanced error handling for edge cases in data processing and transformations.

* **Tests**
  * Expanded test documentation with comprehensive parameter type specifications.

### CODE REVIEW

File: src/Drupal/DrupalExtension/Context/BatchContext.php [hunk 1]
```diff
@@ -54,10 +54,13 @@ public function thereIsAnItemInTheSystemQueue(TableNode $table): void {
...
     $connection = \Drupal::service('database');
+    /** @var \Drupal\Core\Password\PasswordGeneratorInterface $password_generator */
+    $password_generator = \Drupal::service('password_generator');
+    $data = is_string($fields['data']) ? $fields['data'] : (string) json_encode($fields['data']);
     $query = $connection->insert('queue')
       ->fields([
-        'name' => $fields['name'] ?: \Drupal::service('password_generator')->generate(),
-        'data' => serialize(json_decode((string) $fields['data'])),
+        'name' => $fields['name'] ?: $password_generator->generate(),
+        'data' => serialize(json_decode($data)),
```
[the remark this answers is not in the document]
Comment (AlexSkrypnyk) [reply src/Drupal/DrupalExtension/Context/BatchContext.php]: Fixed in c4f40493. Replaced `?:` with a `($fields['name'] ?? '') !== ''` check so the queue name `"0"` is preserved.

File: src/Drupal/DrupalExtension/Context/MinkContext.php [hunk 2]
```diff
@@ -29,11 +29,11 @@ class MinkContext extends MinkExtension implements TranslatableContext {
...
    *
-   * @return array
+   * @return array<int, string>
    *   List of translation resource paths.
    */
-  public static function getTranslationResources() {
-    return self::getMinkTranslationResources() + glob(__DIR__ . '/../../../../i18n/*.xliff');
+  public static function getTranslationResources(): array {
+    return self::getMinkTranslationResources() + (glob(__DIR__ . '/../../../../i18n/*.xliff') ?: []);
   }
```
[the remark this answers is not in the document]
Comment (AlexSkrypnyk) [reply src/Drupal/DrupalExtension/Context/MinkContext.php:40]: Fixed in c4f40493. Switched to `array_merge()` so the i18n XLIFF files are not dropped when the parent translation list is non-empty.

File: src/Drupal/MinkExtension/ServiceContainer/Driver/BrowserKitFactory.php [hunk 3]
```diff
@@ -62,7 +65,7 @@ public function buildDriver(array $config): Definition {
    * @codeCoverageIgnore
    */
   protected function getCwd(): string {
-    return getcwd();
+    return (string) getcwd();
```
[the remark this answers is not in the document]
Comment (AlexSkrypnyk) [reply src/Drupal/MinkExtension/ServiceContainer/Driver/BrowserKitFactory.php]: Fixed in c4f40493. `getCwd()` now throws a `RuntimeException` when `getcwd()` fails instead of casting `false` to an empty string and silently breaking `DrupalFinder::locateRoot()` downstream.
